### ISSUE TITLE
.jdbc.wrapper.SQLResultSetMetaData.getPrecision

### ISSUE THREAD

**Issue by ssky1** [housepower/ClickHouse-Native-JDBC#117]
### JDBC version in pom.xml
ClickHouse server version 20.3.4

### ClickHouse version
2.0-stable

### Error  logs
```
java.sql.SQLFeatureNotSupportedException
        at com.github.housepower.jdbc.wrapper.SQLResultSetMetaData.getPrecision(SQLResultSetMetaData.java:65)
        at org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$.getSchema(JdbcUtils.scala:228)
        at org.apache.spark.sql.execution.datasources.jdbc.JDBCRDD$.resolveTable(JDBCRDD.scala:64)
        at org.apache.spark.sql.execution.datasources.jdbc.JDBCRelation.<init>(JDBCRelation.scala:113)
        at org.apache.spark.sql.execution.datasources.jdbc.JdbcRelationProvider.createRelation(JdbcRelationProvider.scala:45)
        at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:330)
        at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:152)
        at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:125)
        at test_spark.clickhouse_toto_0_1.CLICKHOUSE_TOTO.tJDBCInput_1Process(CLICKHOUSE_TOTO.java:1110)
        at test_spark.clickhouse_toto_0_1.CLICKHOUSE_TOTO.run(CLICKHOUSE_TOTO.java:1586)
        at test_spark.clickhouse_toto_0_1.CLICKHOUSE_TOTO.runJobInTOS(CLICKHOUSE_TOTO.java:1391)
        at test_spark.clickhouse_toto_0_1.CLICKHOUSE_TOTO.main(CLICKHOUSE_TOTO.java:1274)
```

### Steps to reproduce
I am trying to read a click house distributed table from a talend job (tJdbcInput component) 
using ClickHouse-Native-JDBC driver 

The component is trying to read table metadata 
column type, column length and column precision 

I saw in the driver sources that the method is not implemented yet !

so clickhouse can't be used with talend ?

thanks

**Comment by sundy-li** [comment]
Sorry, currently metadata API is not supported.

**Comment by ico01** [comment]
you should use try catch , and ignore the exceptioin

try {
      val pro = new java.util.Properties
      pro.put("driver", ckDriver)
      peopleDFCsv.write
        .mode(SaveMode.Append)
        .option("batchsize", "20000")
        .option("isolationLevel", "NONE")
        .option("numPartitions", "1")
        .jdbc(ckUrl, table, pro)
    } catch {
      // 这里注意下，spark里面JDBC datasource用到的一些获取元数据的方法插件里并没有支持，比如getPrecision & setQueryTimeout等等，都会抛出异常，但是并不影响写入
      case e: SQLFeatureNotSupportedException =>
        println("catch and ignore!")
    }
    spark.close()
  }

https://www.jianshu.com/p/6670fde7e083

**Comment by asd51731** [comment]
> Sorry, currently metadata API is not supported.

if I want to fix the problem, i should fix this native-jdbc code or spark code? Can you give me some help. Thanks

**Comment by sundy-li** [comment]
Hi, You can modify the  native  jdbc  code  of `ClickHouseResultSetMetaData`, override this method as an empty implemention.j

**Comment by pan3793** [comment]
@ssky1 @ico01 This issue should be fixed in `2.3-stable`, would you please have a try?

**Comment by pan3793** [comment]
Close issue since it should be fixed, if not, feel free to reopen it.
